sql/src/main/java/io/crate/metadata/cluster/AlterTableClusterStateExecutor.java

```diff
@@ -0,0 +1,437 @@
+import org.elasticsearch.common.collect.MapBuilder;
+import org.elasticsearch.common.compress.CompressedXContent;
+import org.elasticsearch.common.regex.Regex;
+import org.elasticsearch.common.settings.IndexScopedSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.DeprecationHandler;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.common.xcontent.json.JsonXContent;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.mapper.MapperService;
+import org.elasticsearch.indices.IndicesService;
+import org.elasticsearch.indices.InvalidIndexTemplateException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.index.IndexSettings.same;
+
+public class AlterTableClusterStateExecutor extends DDLClusterStateTaskExecutor<AlterTableRequest> {
+
+    private static final IndicesOptions INDICES_OPTIONS = IndicesOptions.fromOptions(
+        true, true, true, true);
+
+    private final MetaDataMappingService metaDataMappingService;
+    private final IndicesService indicesService;
+    private final AllocationService allocationService;
+    private final IndexNameExpressionResolver indexNameExpressionResolver;
+    private final IndexScopedSettings indexScopedSettings;
+    private final MetaDataCreateIndexService metaDataCreateIndexService;
+
+    public AlterTableClusterStateExecutor(MetaDataMappingService metaDataMappingService,
+                                          IndicesService indicesService,
+                                          AllocationService allocationService,
+                                          IndexScopedSettings indexScopedSettings,
+                                          IndexNameExpressionResolver indexNameExpressionResolver,
+                                          MetaDataCreateIndexService metaDataCreateIndexService) {
+        this.metaDataMappingService = metaDataMappingService;
+        this.indicesService = indicesService;
+        this.indexScopedSettings = indexScopedSettings;
+        this.allocationService = allocationService;
+        this.indexNameExpressionResolver = indexNameExpressionResolver;
+        this.metaDataCreateIndexService = metaDataCreateIndexService;
+    }
+
+    @Override
+    protected ClusterState execute(ClusterState currentState, AlterTableRequest request) throws Exception {
+        Index[] concreteIndices = indexNameExpressionResolver.concreteIndices(currentState,
```

Comment:
Could we move this call elsewhere given that in some cases it becomes unused? (e.g. if partitioned and partionedIndexName != null then we don't use the result here)